Modules/_asynciomodule.c

```diff
@@ -2015,6 +2033,54 @@ leave_task(asyncio_state *state, PyObject *loop, PyObject *task)
     return _PyDict_DelItem_KnownHash(state->current_tasks, loop, hash);
 }
 
+static PyObject *
+swap_current_task(asyncio_state *state, PyObject *loop, PyObject *task)
+{
+    PyObject *prev_task;
+    Py_hash_t hash;
+    hash = PyObject_Hash(loop);
+    if (hash == -1) {
+        return NULL;
+    }
+
+    prev_task = _PyDict_GetItem_KnownHash(state->current_tasks, loop, hash);
+    if (prev_task == NULL) {
+        if (PyErr_Occurred()) {
+            return NULL;
+        }
+        prev_task = Py_None;
+    }
+
+    if (task == Py_None) {
+        if (_PyDict_DelItem_KnownHash(state->current_tasks, loop, hash) == -1) {
+            return NULL;
+        }
+    } else {
+        if (_PyDict_SetItem_KnownHash(state->current_tasks, loop, task, hash) == -1) {
+            return NULL;
+        }
+    }
+
+    Py_INCREF(prev_task);
+
+    return prev_task;
+}
+
+static int
+is_loop_running(PyObject *loop)
+{
+    PyObject *func = PyObject_GetAttr(loop, &_Py_ID(is_running));
+    if (func == NULL) {
+        PyErr_Format(PyExc_TypeError, "Loop missing is_running()");
```

Comment:
It would be much better to take advantage of load method optimization here and avoid creating bound method here.

Reply:
(I.e., use `PyObject_CallMethodNoArgs()`)